rpc/eth_rpc_funcs.go

```diff
@@ -0,0 +1,141 @@
+	ltypes "github.com/loomnetwork/go-loom/types"
+	"github.com/loomnetwork/loomchain/auth"
+	"github.com/loomnetwork/loomchain/rpc/eth"
+	"github.com/loomnetwork/loomchain/vm"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// TODO: This doesn't belong in this package, should be in rpc/eth, but moving it there is non-trivial
+//       due to import cycles
+type SendRawTransactionPRCFunc struct {
+	eth.HttpRPCFunc
+	chainID     string
+	broadcastTx func(tx types.Tx) (*ctypes.ResultBroadcastTx, error)
+}
+
+// Tendermint handlers need parameters translated.
+// Only one method supported.
+func NewSendRawTransactionRPCFunc(chainID string, broadcastTx func(tx types.Tx) (*ctypes.ResultBroadcastTx, error)) eth.RPCFunc {
+	return &SendRawTransactionPRCFunc{
+		chainID:     chainID,
+		broadcastTx: broadcastTx,
+	}
+}
+
+func (t *SendRawTransactionPRCFunc) UnmarshalParamsAndCall(
+	input eth.JsonRpcRequest, conn *websocket.Conn,
+) (json.RawMessage, *eth.Error) {
+	if len(input.Params) == 0 {
+		return nil, eth.NewError(eth.EcInvalidParams, "Parse params", "expected one or more parameters")
+	}
+
+	paramsBytes := []json.RawMessage{}
+	if err := json.Unmarshal(input.Params, &paramsBytes); err != nil {
+		return nil, eth.NewError(eth.EcParseError, "Parse params", err.Error())
+	}
+
+	var data string
+	if err := json.Unmarshal(paramsBytes[0], &data); err != nil {
+		return nil, eth.NewErrorf(eth.EcParseError, "Parse params", "failed to unmarshal input: %v", err)
+	}
+
+	txBytes, err := eth.DecDataToBytes(eth.Data(data))
+	if err != nil {
+		return nil, eth.NewErrorf(
+			eth.EcParseError, "Parse params", "failed to convert input %v to bytes: %v", data, err,
+		)
+	}
+
+	tmTx, err := ethereumToTendermintTx(t.chainID, txBytes)
+	if err != nil {
+		return nil, eth.NewError(
+			eth.EcServer, fmt.Sprintf("failed to wrap eth tx: %v", err), "",
+		)
+	}
+
+	r, err := t.broadcastTx(tmTx)
+	if err != nil {
+		return nil, eth.NewErrorf(eth.EcServer, "Server error", "transaction returned error %v", err)
+	}
+	if r == nil {
+		return nil, eth.NewError(eth.EcServer, "Server error", "transaction returned nil result")
+	}
+	if r.Code != 0 {
+		return nil, eth.NewError(eth.EcServer, fmt.Sprintf("CheckTx failed: %s", r.Log), "")
+	}
+
+	var result json.RawMessage
+	result, err = json.Marshal(eth.EncBytes(r.Hash))
+	if err != nil {
+		return nil, eth.NewErrorf(eth.EcServer, "Server error", "failed to marshal tx hash: %v", err)
+	}
+	return result, nil
+}
+
+// Wraps a raw Ethereum tx in a Loom SignedTx
+func ethereumToTendermintTx(chainID string, txBytes []byte) (types.Tx, error) {
+	msg := &vm.MessageTx{}
+	msg.Data = txBytes
+	var tx etypes.Transaction
+	if err := rlp.DecodeBytes(txBytes, &tx); err != nil {
+		return nil, err
+	}
+
+	if tx.To() != nil {
+		msg.To = loom.Address{
+			ChainID: chainID,
+			Local:   tx.To().Bytes(),
+		}.MarshalPB()
+	}
+
+	ethChainID, err := evmcompat.ToEthereumChainID(chainID)
```

Comment:
As ethChainId only ever depends on `SendRawTransactionPRCFunc.chainID` can we save in  `SendRawTransactionPRCFunc` rather than calculate every time?

Reply:
Yeah, you're right, I'll fix that

Reply:
Also ethSigner := etypes.NewEIP155Signer(ethChainID) couple lines below